audioread/gstdec.py

```diff
@@ -311,7 +311,11 @@ def _new_sample(self, sink):
             # New data is available from the pipeline! Dump it into our
             # queue (or possibly block if we're full).
             buf = sink.emit('pull-sample').get_buffer()
-            self.queue.put(buf.extract_dup(0, buf.get_size()))
+            mem = buf.get_all_memory()
```

Comment:
It would be awesome to put a comment here about why we're using this strategy to copy the data instead of `extract` or `extract_dup`—that way, posterity will not not get confused and try to change it back. :smiley: